application-analytics-default/src/main/java/com/xwiki/analytics/script/AnalyticsScriptService.java

```diff
@@ -57,6 +61,7 @@ public class AnalyticsScriptService implements ScriptService
      * @param parameters a map of the parameters needed for this request
      * @return response from Matomo API, in a normalized JSON format
      */
+    @Unstable
     public JsonNode getMatomoRequestResult(Map<String, String> parameters, Map<String, String> filters,
```

Comment:
The script service is named ``$services.analytics`` not ``$services.matomo``, so I would have expected it to be generic. Can't you rewrite this as a generic method?

Reply:
Hello @mflorea!  From our discussion with Stefana a while ago, we decided that making the app universally compatible with multiple analytics services would be too complex. We've opted to focus solely on Matomo, despite the 'analytics' space name. If you believe a more general approach is better, I think I could introduce a new public method instead of getMatomoRequestResult. This method, called makeRequest, would take a Map<String, String> for parameters and filters, a String for jsonNormaliserHint, and a String for serviceProvider, and it would delegate to the appropriate manager.

Reply:
I think it would be nice to not expose the Matomo implementation in the public APIs, even if we don't plan to have another implementation based on a different analytics service / provider. This will allow us to switch more easily to a different provider, in the future, if we'll have to.

Reply:
I've updated the API to make it more generic by removing the getMatomoRequestResult function and adding a new one called makeRequest. This new function will delegate the request to the appropriate manager.